Re-design paper/reviews page

thomasbrus opened the pull request:
Ben zo goed als klaar met de paper/reviews pagina. D.w.z. de belangrijkste elementen zijn aanwezig.

Wat bewust (nog) niet gedaan is:
- Comments op meerdere niveaus. Denk dat het daar mee te ingewikkeld wordt, en "stackoverflow heeft het ook niet"
- Datum bij review. Dit kan verwerkt worden in de naam/universiteit boven de review
- Star rating boven formulier nieuwe review. Is nu gedaan met jQuery, maar deze past niet bij de rest qua stijl. Hier wil ik later nog wel even naar kijken, zou niet al te moeilijk moeten zijn met een hidden input en wat javascript.
- Optie voor externe of semi-anonieme review. In de huidige situatie is het onduidelijk wat elke optie betekent. De enige optie die écht niet kan ontbreken is anoniem maken. Van semi-anoniem begrijp ik de bedoeling niet helemaal; extern kun je ook aangeven in de tekst van de review zelf.

Omdat in deze branch de nodig dingen zijn gewijzigd qua css, ga ik hier ook in verder voor het in orde maken van de overige pagina's. _Het zou fijn zijn als jullie ondertussen een begin willen maken aan het dynamisch maken van deze pagina. Dat lijkt me het meest efficient:)_ (Even taken verdelen?)

thomasbrus commented:
![openreview-2](https://cloud.githubusercontent.com/assets/232034/2724515/671d50b0-c5a2-11e3-8c06-467a30882a03.png)

thomasbrus commented:
![openreview-1](https://cloud.githubusercontent.com/assets/232034/2724519/7235bdca-c5a2-11e3-829a-ee480d9d83d6.png)

shsdv commented:
Het design ziet er goed uit!

> Het zou fijn zijn als jullie ondertussen een begin willen maken aan het dynamisch maken van deze pagina. Dat lijkt me het meest efficient:) (Even taken verdelen?)

Welke dingen moeten allemaal dynamisch worden? Welke taken zijn er te verdelen?

martijnbastiaan commented:
_Nice_

> Comments op meerdere niveaus. Denk dat het daar mee te ingewikkeld wordt, en "stackoverflow heeft het ook niet"

Is het moeilijk om een speciale "dedicated" pagina daarvoor te maken? Ik snap dat het snel onoverzichtelijk wordt als het allemaal op de paper pagina komt, maar een knop 'discuss this review' die een Reddit achtige pagina start zou niet te lastig moeten zijn denk ik. Alle code is er al (`get_tree`, oude templates). Het bediscussiëren van reviews lijkt me juist een van de sterke punten van OR!

> Optie voor externe of semi-anonieme review. In de huidige situatie is het onduidelijk wat elke optie betekent. De enige optie die écht niet kan ontbreken is anoniem maken. Van semi-anoniem begrijp ik de bedoeling niet helemaal; extern kun je ook aangeven in de tekst van de review zelf.

De uitleg moet inderdaad verwerkt worden in de UI, maar alle opties zijn strikt noodzakelijk:
- public: poster voor iedereen zichtbaar
- semi-anonymous: poster onzichtbaar, maar aanwezig in de database (review/comment telt dus mee voor reputatie)
- anonymous: systeem slaat niet op wie het bericht gepost heeft. Dit garandeert anonimiteit bij het uitlekken van de database, maar zorgt ervoor dat de review niet meetelt voor de reputatie.
- external: telt niet mee voor reputatie, UI geeft hint dat het een externe review is.

thomasbrus commented:
> Welke dingen moeten allemaal dynamisch worden? Welke taken zijn er te verdelen?

In principe kun je het allemaal afleiden uit `paper.html`. Hier even een lijstje:
- [ ] Abstract (elke alinea in een `<p>`, mag ongeveer 140px hoog zijn). Als dit problemen oplevert moet je 't even zeggen.
- [ ] Authors, publication date
- [ ] Keywords (met links, mag '#' zijn)
- [ ] Aantal reviews weergeven (dus zonder comments)
- [ ] Aantal votes
- [ ] Herstellen van JS die up en downvoten regelt en de vote count updatet.
- [ ] Naam van auteur in juiste formaat + universiteit (?)
- [ ] Aantal sterren
- [ ] Review text (newlines als `<br />` als dat lukt)
- [ ] Comments weergeven (geen comments op comments dus?)
- [ ] Comments toevoegen via AJAX.
- [ ] Formulier om een review toe te voegen, kijk ik zelf even naar (ook ivm Martijn's opmerking)

thomasbrus commented:
> Is het moeilijk om een speciale "dedicated" pagina daarvoor te maken? Ik snap dat het snel onoverzichtelijk wordt als het allemaal op de paper pagina komt, maar een knop 'discuss this review' die een Reddit achtige pagina start zou niet te lastig moeten zijn denk ik. Alle code is er al (get_tree, oude templates). Het bediscussiëren van reviews lijkt me juist een van de sterke punten van OR!

Dat vind ik wel een goed idee. Ook omdat er dan iets meer de mogelijkheid is om in discussie te gaan. De vraag is dan alleen wat we laten zien op deze pagina. Je zou bijv alleen level-0 comments kunnen laten zien. (En daar onder idd iets van 'read full discussion')

> De uitleg moet inderdaad verwerkt worden in de UI, maar alle opties zijn strikt noodzakelijk: 

Oke, daar ga ik nog even wat voor bedenken dan. Misschien ook een idee op het op te splitsen in visibility: (public|semi-..|anonymous) en iets van author: (me|external). En daar dan twee drop downs met korte uitlegtekst van te maken.

martijnbastiaan commented:
> Misschien ook een idee op het op te splitsen in visibility: (public|semi-..|anonymous) en iets van author: (me|external). En daar dan twee drop downs met korte uitlegtekst van te maken.

Klinkt goed. Laat even weten wat je doet, dan pas ik eventueel de backend aan (?).

thomasbrus commented:
![schermafbeelding 2014-04-18 om 17 05 37](https://cloud.githubusercontent.com/assets/232034/2742929/b1062aa2-c70b-11e3-8d0a-789262185953.png)
![schermafbeelding 2014-04-18 om 17 10 01](https://cloud.githubusercontent.com/assets/232034/2742930/b1076872-c70b-11e3-8739-373f7229ae2f.png)

martijnbastiaan commented:
Even voor de duidelijkheid: deze ga ik nog niet mergen (niet erg voor planning) door het falen van de tests. Mits alle functionaliteit er is (dwz, de javascript het doet) wil ik hier morgen wel naar kijken.

wietze committed:
Implemented related papers (by authors and keywords/categories) on paper page

wietze committed:
Now hiding 'related' box when no related papers exist

basvandenbrink committed:
Full name fields added to registration form

basvandenbrink committed:
Fixed title and university fields, fixes #109

thomasbrus commented:
(This branch is currently ignored by Travis (9f26a6daf71f7d3913004e4a7e07d148dd36effe))

basvandenbrink committed:
SettingsForm does not extend RegistrationForm anymore

Commit:
Sanified CSS rule scoping

Commit:
Fixed Review API bug, which caused 'visibility' parameter to be ignored in PATCH requests

Commit:
Unified reviews and comments in CSS and HTML, enabled editing

Commit:
Added caching for Review.get_tree_size()

Commit:
Merge branch 'base-design' of github.com:open-review/open-review into base-design

Commit:
Fixed bug causing MathJax to be disabled after forced browser refresh (ctrl+f5).

This changes the behaviour of rendered elements: only DOM elements with a class 'tex' will be rendered on startup.

Commit:
Manually added template directories to settings.py to PyCharm can find them (thus providing hints)

Commit:
Restored discussion tree

Commit:
Fixed spelling issue (seperator --> separator)

Commit:
Added urlize to review rendering

Commit:
Restored reply functionality

Commit:
Fixed stupid bug causing star ratings to be submitted in wrong form

Commit:
Reintroduced deleting

Commit:
VoteView does not exist anymore, removing. Fixes #136.

Commit:
Fixed TestReviewView, removed dead code and documented some non-dead. Fixes #142.

Commit:
Supplying 'text' field to PreviewProcedure is not mandatory anymore, in order to render empty previews. Fixed tests, fixes #126.

Commit:
Input name 'existing' changed to 'new' (for which I can only assume existed a very good reason). Updated tests to support this change. Fixes #138.

Commit:
Merge remote-tracking branch 'origin/feature-registration-form' into base-design

Conflicts:
	openreview/apps/accounts/forms.py

Commit:
Fixed SettingsForm conventions, fixed tests. Fixes #139.

Commit:
Added documentation and wait_for_model to SeleniumTestCase

Commit:
Added error messages for anonymous/deleted voting. Fixes #137.

Commit:
API of login form changed, updated tests to reflect this change. Fixes #129.

Commit:
Form API changed, updated tests to reflect these changes. Fixes #130.

Commit:
Fixed TestDeleteAccountFormLive. Fixes #127.

Commit:
Rewrote TestReviewViewLive. Fixes #135.

Commit:
.new-review changed to .compose-top-level. Changed test to support this change. Fixes #133.

Commit:
Splitted and fixed TestReviewForm. Disabled test_paper_gets_comitted for now. Fixes #131.

Commit:
GET parameter 'p' renamed to 'page'. Updated tests to reflect this change. Fixes #141.

Commit:
Which fool compared integers with 'is'?! Oh wait, it was me. Fixed.

Furthermore: prevent Python integer caching by choosing a high value for COUNTER, thus catching these type of errors earlier.

Commit:
Use X-METHOD-OVERRIDE for deleting things, as not all browsers support HTTP DELETE.

Commit:
Do not test for preview section directly after hitting 'edit' button.

Commit:
Removed old add paper code

Commit:
Merge remote-tracking branch 'origin/feature-related-papers' into base-design

Conflicts:
	openreview/apps/papers/views.py
